src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
 export const App = () => {
+  function getUserById(userId: number) {
+    return usersFromServer.find(user => user.id === userId)
+    || null;
+  }
+
+  const initialTodos = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId),
+  }));
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasTitle, setHasTitle] = useState(false);
+  const [hasUserId, setHasUserId] = useState(false);
+
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const getMaxTodoId = () => {
+    return Math.max(...todos.map(todo => todo.id), 0);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+
+    setHasTitle(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+
+    setHasUserId(false);
+  };
+  //
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitle(!title);
+    setHasUserId(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getMaxTodoId() + 1,
+      title,
+      userId,
+      user: getUserById(userId),
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
```

Comment:
see info: https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state